sdks/java/core/src/main/java/org/apache/beam/sdk/coders/DelegateCoder.java

```diff
@@ -101,8 +104,28 @@ public Object structuralValue(T value) throws Exception {
   }
 
   @Override
+  public boolean equals(Object o) {
+    if (o == null || this.getClass() != o.getClass()) {
```

Comment:
Use instanceof will allow DelegateCoder equal to its subclass, which would break the symmetric property when its subclass override equals().

http://www.artima.com/lejava/articles/equality.html